### DISCUSSION TITLE
Release a new dockerhub image version when providers get an update and version is current

### DISCUSSION THREAD

**Discussion by alejandrofm** [apache/airflow#34965] [Ideas] [answered]
Hi! there are releases, especially on providers, that are not always in line with the released version of Airflow, for example:
Airflow 2.7.1 was released, and the docker image was created.
3 days later a new version of providers was released.
That update will be only available when 2.7.2 release is out, and a new provider version will surely be released.

It will be nice that:
- Airflow and providers and airflow releases were aligned (not ideal)
or
- When a new release/fix on providers gets released, re-generate the docker image so the last airflow version gets the last version of Providers.

Thanks!

**Comment by Taragolis** [comment]
This could turn life of release managers and other users into the hell, right now if image released with specific tag it immutable, and if we change then users might found that version of providers changed which might breaks their pipelines.

I would recommend:
1. Use slim airflow image, see: [Docker Image for Apache Airflow](https://airflow.apache.org/docs/docker-stack/index.html#docker-image-for-apache-airflow) and [extend it](https://airflow.apache.org/docs/docker-stack/build.html#extending-the-image)
2. Build own [custom Airflow image](https://airflow.apache.org/docs/docker-stack/build.html#customizing-the-image)

**Comment by potiuk** [comment, marked as answer]
And just to reiterate - Nope. This is not going to change. 

It's explained why in https://airflow.apache.org/docs/docker-stack/index.html#fixing-images-at-release-time that we are releasing the images at release time and it is a "reference" image based on constraints of Airflow at the moment of release providing "golden" set of dependencies.

But you are absolutely free to extend the image as you wish at the moment providers are released following https://airflow.apache.org/docs/docker-stack/build.html#examples-of-image-extending it's just few lines of docker file and single `docker build` command. Super easy for people like you

**Reply by alejandrofm** [reply]
Makes sense, trying to make things simpler can broke a lot of other things.
Thank you!
